wolfcrypt/src/aes.c

```diff
@@ -10033,16 +10033,16 @@ static WARN_UNUSED_RESULT int roll_auth(
     /* encode the length in */
     if (inSz <= 0xFEFF) {
         authLenSz = 2;
-        out[0] ^= ((inSz & 0xFF00) >> 8);
-        out[1] ^=  (inSz & 0x00FF);
+        out[0] = (byte)((int)out[0]^((inSz & 0xFF00) >> 8));
```

Comment:
Can't you instead do this? `out[0] ^= (byte)((inSz & 0xFF00) >> 8);`? Worried the XOR on int will not behave same.

Reply:
Get error on these anyways:

```
wolfcrypt/src/aes.c:10036:25: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
        out[0] = (byte)((int)out[0]^((inSz & 0xFF00) >> 8));
                        ^~~~~~~~~~~~
wolfcrypt/src/aes.c:10037:25: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
        out[1] = (byte)((int)out[1]^(inSz & 0x00FF));
                        ^~~~~~~~~~~~
wolfcrypt/src/aes.c:10042:25: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
        out[2] = (byte)((int)out[2]^((inSz & 0xFF000000) >> 24));
                        ^~~~~~~~~~~~
wolfcrypt/src/aes.c:10043:25: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
        out[3] = (byte)((int)out[3]^((inSz & 0x00FF0000) >> 16));
                        ^~~~~~~~~~~~
wolfcrypt/src/aes.c:10044:25: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
        out[4] = (byte)((int)out[4]^((inSz & 0x0000FF00) >>  8));
                        ^~~~~~~~~~~~
wolfcrypt/src/aes.c:10045:26: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
        out[5] =  (byte)((int)out[5]^(inSz & 0x000000FF));
                         ^~~~~~~~~~~~
```